metadata/src/main/java/org/apache/kafka/controller/BrokerHeartbeatManager.java

```diff
@@ -61,6 +67,11 @@ static class BrokerHeartbeatState {
          */
         private final int id;
 
+        /**
+         * The directories which the broker currently has available.
+         */
+        private final List<Uuid> directories;
```

Comment:
Can we rename this to `directoriesSorted` or `sortedDirectories` to indicate that this list must be sorted?  We depend on it being sorted when invoking `DirectoryId.isOnline()`.  And when I look at where we invoke that method, it seems that the field `BrokerRegistration.directories` should get a similar rename.

The other alternative is to remove the constraint that the list be sorted within `DirectoryId.isOnline()`.  Since we don't expect to have a large number of directories, I would lean toward this option in order to avoid bugs down the road if we forget to sort a list somewhere.  Could we then remove the sorting of the list in this class and within `BrokerRegistration` and in any tests?  Or am I missing something and we really do need to keep these sorted?

Reply:
The point was to use binary search for lookups, see `DirectoryId.isOnline`. Since we don't expect a large list of directories, it may be ok to always to a O(N) search. We can come back to this in the future if it proves to be an issue. I'll change it to avoid any sorting.

Reply:
Can you just make it sorted? The nondeterminism of unsorted just sucks for testing / debugging / etc.

Reply:
Reverted, back to a sorted list, but now clearly named as such.